chain/network/src/types.rs

```diff
@@ -263,25 +263,48 @@ impl From<HandshakeV2> for Handshake {
     }
 }
 
+/// Contains metadata used for routing messages to particular `PeerId` or `AccountId`.
 #[cfg_attr(feature = "deepsize_feature", derive(DeepSizeOf))]
 #[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Clone, Debug)]
-pub struct SyncData {
-    pub(crate) edges: Vec<Edge>,
-    pub(crate) accounts: Vec<AnnounceAccount>,
+pub struct RoutingTableSync {
+    /// List of known edges from `RoutingTableActor::edges_info`.
+    edges: Vec<Edge>,
+    /// List of known `account_id` to `PeerId` mappings.
+    /// Useful for `send_message_to_account` method, to route message to particular account.
+    accounts: Vec<AnnounceAccount>,
 }
 
-impl SyncData {
-    pub(crate) fn edge(edge: Edge) -> Self {
-        Self { edges: vec![edge], accounts: Vec::new() }
+impl RoutingTableSync {
+    pub(crate) fn new(edges: Vec<Edge>, accounts: Vec<AnnounceAccount>) -> Self {
+        Self { edges, accounts }
+    }
+
+    pub(crate) fn from_edges(edges: Vec<Edge>) -> Self {
+        Self { edges, accounts: Default::default() }
+    }
+
+    pub(crate) fn from_accounts(accounts: Vec<AnnounceAccount>) -> Self {
+        Self { edges: Default::default(), accounts }
     }
```

Comment:
When defining factory functions, I prefer to delegate to make it clear which constructor is primary: 

```suggestion
    pub(crate) fn from_edges(edges: Vec<Edge>) -> Self {
        Self::new(edges, Vec::new())
    }

    pub(crate) fn from_accounts(accounts: Vec<AnnounceAccount>) -> Self {
        Self::new(Vec::new(), accounts)
    }
```

useful to eg, be able to add creation debug prints in one place, and too more clearly see casese where helper factorise do genuienly different things.

Reply:
Thanks, that's a good idea. I also removed `from_edge`, and `from_account` constructors to make code cleaner.